File: redact-dropwizard-example/src/main/java/us/jsy/redact/HelloWorldResource.java
package us.jsy.redact;

import com.codahale.metrics.annotation.Timed;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.QueryParam;
import javax.ws.rs.core.MediaType;
import java.util.Optional;
import java.util.concurrent.atomic.AtomicLong;

@Path("/hello-world")
@Produces(MediaType.APPLICATION_JSON)
public class HelloWorldResource {
    private static final Logger logger = LoggerFactory.getLogger(HelloWorldResource.class);

    private final String template;
    private final String defaultName;
    private final AtomicLong counter;

    public HelloWorldResource(String template, String defaultName) {
        this.template = template;
        this.defaultName = defaultName;
        this.counter = new AtomicLong();
    }

    @GET
    @Timed
    public Saying sayHello(@QueryParam("name") Optional<String> name) {
        logger.debug("name: {}", name);
        final String value = String.format(template, name.orElse(defaultName));

        logger.info("say: {}", value);

        return new Saying(counter.incrementAndGet(), value);
    }
}

File: redact-dropwizard-kotlin-example/src/main/java/us/jsy/redact/HelloWorldResource.kt
package us.jsy.redact

import com.codahale.metrics.annotation.Timed
import org.slf4j.LoggerFactory
import java.util.*
import java.util.concurrent.atomic.AtomicLong
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.QueryParam
import javax.ws.rs.core.MediaType

@Path("/hello-world")
@Produces(MediaType.APPLICATION_JSON)
class HelloWorldResource(private val template: String?, private val defaultName: String?) {
    private val counter: AtomicLong

    @GET
    @Timed
    fun sayHello(@QueryParam("name") name: Optional<String?>): Saying {
        logger.debug("name: {}", name)
        val value = String.format(template!!, name.orElse(defaultName))
        logger.info("say: {}", value)
        return Saying(counter.incrementAndGet(), value)
    }

    companion object {
        private val logger = LoggerFactory.getLogger(HelloWorldResource::class.java)
    }

    init {
        counter = AtomicLong()
    }
}